nixos/modules/services/web-apps/rutorrent.nix

```diff
@@ -0,0 +1,281 @@
+              ln -sf ${rutorrentConfig} ${cfg.home}/conf/config.php
+
+              cp -r ${pkgs.rutorrent}/php ${cfg.home}/
+
+              ${optionalString (cfg.plugins != [])
+                ''cp -r ${concatMapStringsSep " " (p: "${pkgs.rutorrent}/plugins/${p}") cfg.plugins} ${cfg.home}/plugins/''}
+
+              chown -R rutorrent:rutorrent ${cfg.home}/{conf,share,logs,plugins}
+              chmod -R 755 ${cfg.home}/{conf,share,logs,plugins}
+            '';
+            serviceConfig.Type = "oneshot";
+          };
+        };
+
+        tmpfiles.rules = [ "d '${cfg.home}' 0775 rutorrent rutorrent -" ];
+      };
+
+      users.groups.rutorrent = {};
+
+      users.users = {
+        rutorrent = {
+          home = cfg.home;
+          group = "rutorrent";
+          extraGroups = [ config.services.rtorrent.group ];
+          description = "rutorrent Daemon user";
+          isSystemUser = true;
+        };
+
+        "${config.services.rtorrent.user}" = {
+          extraGroups = [ "rutorrent" ];
+        };
+      };
+    }
+
+    (mkIf cfg.nginx.enable (mkMerge [
+      { services = {
+          phpfpm = {
+            pools.rutorrent = {
+              user = "rutorrent";
+              group = "rtorrent";
+              settings = mapAttrs (name: mkDefault) {
+                "listen.owner" = config.services.nginx.user;
+                "listen.group" = config.services.nginx.group;
+              } // cfg.poolSettings;
+              phpEnv."PATH" = with pkgs; lib.makeBinPath [ php curl procps gzip ];
+            };
+          };
+
+          nginx = {
+            enable = true;
+            virtualHosts = {
+              ${cfg.hostName} = {
+                root = cfg.home;
+                locations = {
+                  "~ [^/]\.php(/|$)" = {
+                    extraConfig = ''
+                      fastcgi_split_path_info ^(.+?\.php)(/.*)$;
+                      if (!-f $document_root$fastcgi_script_name) {
+                        return 404;
+                      }
+
+                      # Mitigate https://httpoxy.org/ vulnerabilities
+                      fastcgi_param HTTP_PROXY "";
+
+                      fastcgi_pass unix:${config.services.phpfpm.pools.rutorrent.socket};
+                      fastcgi_index index.php;
+
+                      include ${pkgs.nginx}/conf/fastcgi.conf;
+                    '';
+                  };
+                };
+              };
+            };
+          };
+        };
+      }
+
+      (mkIf cfg.nginx.exposeInsecureRPC2mount {
+        services.nginx.virtualHosts."${cfg.hostName}".locations."/RPC2" = {
+          extraConfig = ''
+            include ${pkgs.nginx}/conf/scgi_params;
+            scgi_pass unix:${cfg.rpcSocket};
+          '';
+        };
+
+        services.rtorrent.configText = ''
+          schedule = scgi_group,0,0,"execute.nothrow=chown,\":nginx\",(cfg.rpcsock)"
```

Comment:
This is dirty but I need a way to rtorrent's rpc socket accessible by nginx if `exposeInsecureRPC2mount` is set. This would get overwritten if someone use `mkForce` to set it's rtorrent's `configText`. Maybe I should use some systemd post start hook to chown the socket.

Reply:
What if we just document this requirement and add an option to configure the socket group to the rtorrent module?

Reply:
Yes, this is better since rutorrent doesn't actually need the rpc mount if it use one of the `rpc` or `httprpc` plugin. Also one could use rtorrent and the rpc mount with some other client (like flood or a mobile client).

Reply:
Btw, something related to this that I noticed in the documentation. It's recommended to add this line to rtorrent so that rutorrent plugins get initialized on startup. Otherwise the plugins don't start until you open the ruTorrent client manually.

```
execute2 = {sh,-c,/usr/bin/php /var/www/ruTorrent/php/initplugins.php tom &}
```

https://github.com/Novik/ruTorrent/wiki/Plugins#starting-plugins-with-rtorrent

Though, this is super annoying for a number of reasons:
1. the user (`tom`) depends on the http auth username
2. rtorrent now depends on rutorrent running

I'm not really sure how to approach this one.

Reply:
I added a way to change `rtorrent`'s socket group and change it if `rutorrent` is enabled. Should I make a separate PR for the `rtorrent`'s change?

For `initplugins.php` the simplest way would be to let the user do it, just adding some documentation about it.

Or I could check if the nginx vhost has auth and add the config line directly to `rtorrent` config if `rutorrent` is enabled.